src/java/org/apache/cassandra/index/sai/disk/vector/VectorMemtableIndex.java

```diff
@@ -203,7 +203,8 @@ public KeyRangeIterator search(QueryContext context, Expression expr, AbstractBo
         float threshold = expr.getEuclideanSearchThreshold();
 
         SortingIterator.Builder<PrimaryKey> keyQueue;
-        try (var pkIterator = searchInternal(context, qv, keyRange, graph.size(), graph.size(), threshold))
+        // Threshold queries do not use pruning.
+        try (var pkIterator = searchInternal(context, qv, keyRange, graph.size(), graph.size(), threshold, false))
```

Comment:
This is a behavior change independent from the new option, correct?

I am fine with it, but I wanted to double check

Reply:
Yes, it is. I mentioned it in the PR description, and arguably, it could have been its own PR, but I think it's pretty minor overall. Threshold search has its own termination logic though, so it's possible it doesn't have an impact.